src/main/java/de/tum/in/www1/artemis/service/iris/websocket/IrisWebsocketDTO.java

```diff
@@ -49,32 +40,32 @@ public IrisWebsocketDTO(@Nullable IrisMessage message, @Nullable Throwable throw
      * This method categorizes the WebSocket message type as follows:
      * <ul>
      * <li>{@link IrisWebsocketMessageType#MESSAGE} if the {@code message} parameter is not null.</li>
-     * <li>{@link IrisWebsocketMessageType#ERROR} if the {@code message} is null and {@code throwable} is not null.</li>
      * <li>{@link IrisWebsocketMessageType#STATUS} if both {@code message} and {@code throwable} are null.</li>
      * </ul>
      *
-     * @param message   The message associated with the WebSocket, which may be null.
-     * @param throwable The throwable associated with the WebSocket, which may also be null.
+     * @param message The message associated with the WebSocket, which may be null.
      * @return The {@link IrisWebsocketMessageType} indicating the type of the message based on the given parameters.
      */
-    private static IrisWebsocketMessageType determineType(@Nullable IrisMessage message, @Nullable Throwable throwable) {
+    private static IrisWebsocketMessageType determineType(@Nullable IrisMessage message) {
         if (message != null) {
             return IrisWebsocketMessageType.MESSAGE;
         }
-        else if (throwable != null) {
-            return IrisWebsocketMessageType.ERROR;
-        }
-        else {
-            return IrisWebsocketMessageType.STATUS;
-        }
+        return IrisWebsocketMessageType.STATUS;
     }
 
     @Override
-    public Map<String, Object> translationParams() {
-        return translationParams != null ? Collections.unmodifiableMap(translationParams) : null;
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        IrisWebsocketDTO that = (IrisWebsocketDTO) o;
+        return type == that.type && Objects.equals(message, that.message) && Objects.equals(rateLimitInfo, that.rateLimitInfo) && Objects.equals(stages, that.stages);
     }
 
     public enum IrisWebsocketMessageType {
-        MESSAGE, STATUS, ERROR
+        MESSAGE, STATUS
```

Comment:
At this point we could also just remove the type entirely, as everything is technically now a STATUS update.

Reply:
Sounds like an awesome task for a follow up